test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,280 @@
+
+const depositAmount = BigNumber.from(100);
+const depositNativeAmount = ethers.utils.parseUnits(depositAmount.toString(), 9);
+const depositArgs = [{ value: depositNativeAmount }] as [{ value: BigNumber }];
+const withdrawAmount = BigNumber.from(100);
+const withdrawArg = [withdrawAmount] as [BigNumber];
+
+// TODO: remove this when poco is also available in Native mode
+if (CONFIG.chains.default.asset === 'Native') {
+    describe('EscrowNative', () => {
+        let proxyAddress: string;
+        let [iexecPoco, , iexecPocoAsAccountA, iexecPocoAsAdmin]: IexecInterfaceNative[] = [];
+        let [iexecAdmin, accountA, accountB, anyone]: SignerWithAddress[] = [];
+
+        beforeEach('Deploy', async () => {
+            proxyAddress = await loadHardhatFixtureDeployment();
+            await loadFixture(initFixture);
+        });
+
+        async function initFixture() {
+            const accounts = await getIexecAccounts();
+            ({ iexecAdmin, anyone: accountA, requester: accountB, anyone } = accounts);
+
+            iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+            iexecPocoAsAccountA = iexecPoco.connect(accountA);
+            iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+        }
+
+        describe('Receive and Fallback', () => {
+            it('Should call receive successfully', async () => {
+                await expect(
+                    accountA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: depositNativeAmount,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-depositNativeAmount, depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should call fallback successfully', async () => {
+                const randomData = ethers.utils.hexlify(
+                    ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+                );
+
+                await expect(
+                    accountA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: depositNativeAmount,
+                        data: randomData,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-depositNativeAmount, depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+        });
+
+        describe('Deposits', () => {
```

Comment:
I think we're missing a test for `Should deposit with zero value`. It would be useful for non-regression.
We can add a TODO since it was not covered in the JS version.

Reply:
Same thing for withdraw.

Reply:
Added todos